src/jaxsim/api/model.py

```diff
@@ -0,0 +1,1099 @@
+        data: The data of the considered model.
+        output_vel_repr:
+            The output velocity representation of the free-floating jacobians.
+
+    Returns:
+        The (nL, 6, 6+dofs) array containing the stacked free-floating
+        jacobians of the links. The first axis is the link index.
+    """
+
+    if output_vel_repr is None:
+        output_vel_repr = data.velocity_representation
+
+    # The body frame of the Link.jacobian method is the link frame L.
+    # In this method, we want instead to use the base link B as body frame.
+    # Therefore, we always get the link jacobian having Inertial as output
+    # representation, and then we convert it to the desired output representation.
+    match output_vel_repr:
+        case VelRepr.Inertial:
+            to_output = lambda J: J
+
+        case VelRepr.Body:
+
+            def to_output(W_J_Wi):
+                W_H_B = data.base_transform()
+                B_X_W = jaxlie.SE3.from_matrix(W_H_B).inverse().adjoint()
+                return B_X_W @ W_J_Wi
+
+        case VelRepr.Mixed:
+
+            def to_output(W_J_Wi):
+                W_H_B = data.base_transform()
+                W_H_BW = jnp.array(W_H_B).at[0:3, 0:3].set(jnp.eye(3))
+                BW_X_W = jaxlie.SE3.from_matrix(W_H_BW).inverse().adjoint()
+                return BW_X_W @ W_J_Wi
+
+        case _:
+            raise ValueError(output_vel_repr)
+
+    # Get the link jacobians in Inertial representation and convert them to the
+    # target output representation in which the body frame is the base link B
+    J_free_floating = jax.vmap(
+        lambda i: to_output(js.link.jacobian(model=model, data=data, link_index=i))
+    )(jnp.arange(model.number_of_links()))
+
+    return J_free_floating
+
+
+@functools.partial(jax.jit, static_argnames=["prefer_aba"])
+def forward_dynamics(
+    model: JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    joint_forces: jtp.VectorLike | None = None,
+    external_forces: jtp.MatrixLike | None = None,
+    prefer_aba: float = True,
+) -> tuple[jtp.Vector, jtp.Vector]:
+    """
+    Compute the forward dynamics of the model.
+
+    Args:
+        model: The model to consider.
+        data: The data of the considered model.
+        joint_forces:
+            The joint forces to consider as a vector of shape `(dofs,)`.
+        external_forces:
+            The external forces to consider as a matrix of shape `(nL, 6)`.
+        prefer_aba: Whether to prefer the ABA algorithm over the CRB one.
+
+    Returns:
+        A tuple containing the 6D acceleration in the active representation of the
+        base link and the joint accelerations resulting from the application of the
+        considered joint forces and external forces.
+    """
+
+    forward_dynamics_fn = forward_dynamics_aba if prefer_aba else forward_dynamics_crb
+
+    return forward_dynamics_fn(
+        model=model, data=data, tau=joint_forces, external_forces=external_forces
+    )
```

Comment:
`tau` is not present as a keyword argument in both ABA and CRB, typo?
```suggestion
    return forward_dynamics_fn(
        model=model, data=data, joint_forces=joint_forces, external_forces=external_forces
    )
```